schemas/stsci.edu/asdf/0.1.0/core/column.yaml

```diff
@@ -0,0 +1,47 @@
+%YAML 1.1
+---
+$schema: "http://stsci.edu/schemas/yaml-schema/draft-01"
+id: "http://stsci.edu/schemas/asdf/0.1.0/core/column"
+tag: "tag:stsci.edu:asdf/0.1.0/core/column"
+
+title: >
+  A column in a table.
+
+description: |
+  Each column contains a name and an array of data, and an optional description
+  and unit.
+
+type: object
+properties:
+  name:
+    description: The name of the column
+    type: string
+    pattern: "[A-Za-z_][A-Za-z0-9_]*"
+
+  data:
+    description: |
+      The array data for the column.  The array must have a scalar
+      datatype (that is, it may not be a structured array).
```

Comment:
I'm not necessarily talking about general recarrays (i.e. structs).  Just columns where the values are, for example, 5x5 arrays.  That *is* supported by `Table` (and in fact was on the of the drivers for its creation).  

I don't think recarray columns make a whole lot of sense, but maybe someone could come up with a reason for it.  Fine with leaving it out for now.